sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -95,9 +77,98 @@ case class CreateHiveTableAsSelectCommand(
     Seq.empty[Row]
   }
 
+  def getDataWritingCommand(
```

Comment:
I feel it's better to have 2 methods: `writingCommandForExistingTable`, `writingCommandForNewTable`